CMakeLists.txt

```diff
@@ -416,6 +452,13 @@ foreach(target
     # Ensure NANOARROW_DEBUG is defined for debug builds
     target_compile_definitions(${target} PUBLIC "$<$<CONFIG:Debug>:NANOARROW_DEBUG>")
 
+    # Ensure NANOARROW_DLL is set when building and linking to the library
+    if(target MATCHES "_shared$")
```

Comment:
Targets have a `TYPE` property that you could check here instead of the name: https://cmake.org/cmake/help/latest/prop_tgt/TYPE.html